application-task-manager-default/src/main/java/com/xwiki/taskmanager/internal/TaskMacroUpdateEventListener.java

```diff
@@ -0,0 +1,219 @@
+            }
+            context.put(TASK_UPDATE_FLAG, true);
+            deleteTaskPages(document, context, previousDocTasks);
+            createOrUpdateTaskPages(document, context, tasks);
+            context.put(TASK_UPDATE_FLAG, null);
+        }
+    }
+
+    private void deleteTaskPages(XWikiDocument document, XWikiContext context, List<Task> previousDocTasks)
+    {
+        for (Task previousDocTask : previousDocTasks) {
+            try {
+                XWikiDocument taskDoc = context.getWiki().getDocument(previousDocTask.getReference(), context);
+                BaseObject taskObj = taskDoc.getXObject(TASK_CLASS_REFERENCE);
+                if (!document.getDocumentReference()
+                    .equals(resolver.resolve(taskObj.getLargeStringValue(Task.OWNER))))
+                {
+                    continue;
+                }
+                if (authorizationManager.hasAccess(Right.DELETE, context.getUserReference(),
+                    previousDocTask.getReference()))
+                {
+                    context.getWiki().deleteDocument(taskDoc, context);
+                } else if (authorizationManager.hasAccess(Right.EDIT, context.getUserReference(),
+                    previousDocTask.getOwner()))
+                {
+                    taskObj.set(Task.OWNER, "", context);
+                    context.getWiki().saveDocument(taskDoc, context);
+                } else {
+                    logger.warn(
+                        "The task macro with id [{}] was removed but the associated page could not be deleted or "
+                            + "modified.",
+                        previousDocTask.getReference());
+                }
+            } catch (XWikiException e) {
+                logger.error("Failed to remove the Task Document with id [{}]", previousDocTask.getReference());
+            }
+        }
+    }
+
+    private void createOrUpdateTaskPages(XWikiDocument document, XWikiContext context, List<Task> tasks)
+    {
+        for (Task task : tasks) {
+            DocumentReference macroReference = task.getReference();
+            try {
+                if (!authorizationManager.hasAccess(Right.EDIT, context.getUserReference(), macroReference)) {
+                    logger.warn(
+                        "The user [{}] edited the macro with id [{}] but does not have edit rights over it's "
+                            + "corresponding page.",
+                        context.getUserReference(), macroReference);
+                    continue;
+                }
+
+                XWikiDocument taskDoc = context.getWiki().getDocument(macroReference, context).clone();
+
+                BaseObject taskObj = taskDoc.getXObject(TASK_CLASS_REFERENCE, true, context);
+
+                if (!taskDoc.isNew() && !document.getDocumentReference()
+                    .equals(resolver.resolve(taskObj.getLargeStringValue(Task.OWNER))))
```

Comment:
```suggestion
                    .equals(resolver.resolve(taskObj.getLargeStringValue(Task.OWNER), taskReference)))
```

Same as above: the owner reference should be resolved relative to the task reference.